src/Components/NavbarHorizontal/PersonalTools.php

```diff
@@ -243,21 +248,67 @@ protected function getDropdownToggle(): string {
 
 		$newtalkNotifierHtml = $this->getNewtalkNotifier();
 		$userNameHtml = $this->getUserName();
+
 		MediaWikiServices::getInstance()->getHookContainer()->run( 'ChameleonNavbarHorizontalPersonalToolsLinkInnerHtml',
 			[ &$newtalkNotifierHtml, &$userNameHtml, $this ] );
 
 		$this->indent( 1 );
 
-		$dropdownToggle = IdRegistry::getRegistry()->element( 'a', [ 'class' => $toolsClass,
+		$attr = [
+			'class' => $toolsClass,
 			'href' => '#', 'data-toggle' => 'dropdown', 'data-boundary' => 'viewport',
-			'title' => $toolsLinkText ], $newtalkNotifierHtml . $userNameHtml,
+			'title' => $toolsLinkText
+		];
+		
+		if ( $this->avatarUrl ) {
+			$attr['style'] = "background-image:url('$this->avatarUrl')";
+		}
+		
+		$dropdownToggle = IdRegistry::getRegistry()->element( 'a', $attr, $newtalkNotifierHtml . $userNameHtml,
 			$this->indent() );
 
 		$this->indent( -1 );
 
 		return $dropdownToggle;
 	}
 
+	private function setUserAvatar() {		
+		if ( !empty( $GLOBALS['chameleonDisableAvatar'] ) ) {
```

Comment:
I merged the code, but this is not the right approach:
* This should be a setting/attribute on the component in the XML. It is not a global Chameleon setting, but rather specific to one component.
* It should be opt-in (avatars should be disabled by default), otherwise existing wikis might suddenly see avatars if they coincidentally had images with the matching names. This causes an unexpected change.

Reply:
Fixed in https://github.com/ProfessionalWiki/chameleon/pull/432